erpnext-fhir-mapper/src/main/java/com/ozonehis/eip/mappers/erpnext/AddressMapper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.mappers.erpnext;
+
+import com.ozonehis.eip.mappers.ToERPNextMapping;
+import com.ozonehis.eip.model.erpnext.Address;
+import java.util.ArrayList;
+import java.util.List;
+import org.hl7.fhir.r4.model.Element;
+import org.hl7.fhir.r4.model.Extension;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AddressMapper implements ToERPNextMapping<org.hl7.fhir.r4.model.Address, Address> {
+
+    private static final String ADDRESS_TYPE = "Personal";
+
+    private static final String ADDRESS_EXTENSION_URL = "http://fhir.openmrs.org/ext/address";
+
+    private static final String ADDRESS1_EXTENSION = "http://fhir.openmrs.org/ext/address#address1";
+
+    private static final String ADDRESS2_EXTENSION = "http://fhir.openmrs.org/ext/address#address2";
+
+    @Override
+    public Address toERPNext(org.hl7.fhir.r4.model.Address fhirAddress) {
+        Address address = new Address();
+        address.setAddressName(fhirAddress.getIdElement().getValue());
+        address.setCity(fhirAddress.getCity());
+        address.setCountry(fhirAddress.getCountry());
+        address.setPostalCode(fhirAddress.getPostalCode());
+        address.setState(fhirAddress.getState());
+        address.setPrimaryAddress(true);
```

Comment:
We do have a kind of hack in the FHIR2 module to indicate someone's primary address (preferred address in OMRS terms): the resource should have the `Address.use` field set to `home`. Since patients can have multiple addresses, this may help avoid any data validation errors.

Reply:
In the `PatientProcessor` class, when processing a patient's data, only the address marked as 'HOME' is considered. This means that only the preferred address of the patient is synced to the ERPNext.